pkg/pgmodel/ingestor/duplicates.go

```diff
@@ -0,0 +1,49 @@
+package ingestor
+
+import (
+	"time"
+
+	"go.uber.org/atomic"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgmodel/metrics"
+)
+
+const reportDuplicatesInterval = time.Second * 10
+
+var (
+	effectedMetrics             atomic.Uint64
+	isReportDuplicatesActive    atomic.Bool
+	pendingReportableDuplicates atomic.Bool
+)
+
+func init() {
+	isReportDuplicatesActive.Store(false)
+	pendingReportableDuplicates.Store(false)
+}
+
+func registerDuplicates(count int64) {
+	metrics.DuplicateSamples.Add(float64(count))
+	metrics.DuplicateWrites.Inc()
+}
+
+func reportDuplicates(duplicates uint64) {
+	effectedMetrics.Store(duplicates)
```

Comment:
again not sure what this number says or how to interpret it. since this is a store not a sum. I'd actually expect this to be a sum of the /samples/ seen since last reporting. Or a sum of the table inserts (metrics) that had this since last reporting.

Reply:
Yeah, better to store this as a counter (i.e., `metrics.AffectedMetrics`. But, the `effectedMetrics` needs to be Store since it's used in the comparison.